### ISSUE TITLE
Switch over to using torch.testing.assert_close in tests

### ISSUE THREAD

**Issue by epwalsh** [allenai/allennlp#5322]
`assert_allclose` is being deprecated. See https://github.com/pytorch/pytorch/issues/61844.

**Comment by dirkgr** [comment]
The suggested replacement was only added in torch 1.9.0, so we need to wait until that is the oldest version we support before we tackle this.

**Comment by epwalsh** [comment]
I don't think this is a user-facing change but let me know if I'm wrong. Because as far as I know we only use this in internal tests, and we only run those tests against the latest version of PyTorch.
